### PR TITLE
#706 - Socket::read() adds PHP_NORMAL_READ and PHP_BINARY_READ

### PR DESCRIPTION
**Pull request by SirSnyder**

### CODE REVIEW

File: classes/socket.h [hunk 1]
```diff
@@ -237,12 +237,13 @@ PHP_METHOD(Socket, select) {
 PHP_METHOD(Socket, read) {
 	zend_long length = 0;
 	zend_long flags = 0;
+        zend_long type = PHP_BINARY_READ;
```
Comment (tpunt) [inline classes/socket.h]: nit: this should use tabs for indentation.

File: php_pthreads.c [hunk 2]
```diff
@@ -324,7 +324,7 @@ PHP_MINIT_FUNCTION(pthreads)
 	INIT_CLASS_ENTRY(ce, "Socket", pthreads_socket_methods);
 	pthreads_socket_entry = zend_register_internal_class_ex(&ce, pthreads_threaded_entry);
 	pthreads_socket_entry->create_object = pthreads_socket_ctor;
-
+
```
Comment (tpunt) [inline php_pthreads.c]: nit: unnecessary whitespace change.

**Comment by SirSnyder** [comment]
@tpunt, new editor with working indentation ;-) files updated

File: src/socket.h [hunk 3]
```diff
@@ -30,6 +30,9 @@ typedef struct _pthreads_socket_t {
 	zend_bool blocking;
 } pthreads_socket_t;
 
+#define PHP_NORMAL_READ 0x0001
+#define PHP_BINARY_READ 0x0002
```
Comment (dktapps) [inline src/socket.h]: you might want to define Socket class constants for these. These won't be available as userland constants if ext/sockets isn't available.

**Comment by SirSnyder** [comment]
Follow up on https://github.com/krakjoe/pthreads/pull/828
